docs/orchestrating-elastic-stack-applications/elasticsearch/autoscaling.asciidoc

```diff
@@ -0,0 +1,197 @@
+:parent_page_id: elasticsearch-specification
+:page_id: autoscaling
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{parent_page_id}.html#k8s-{page_id}[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-{page_id}"]
+= Elasticsearch autoscaling
+
+experimental[]
+
+ECK can leverage the autoscaling API introduced in Elasticsearch 7.11 to adjust automatically the number of Pods and the allocated resources in a tier. Currently, autoscaling is supported for Elasticsearch link:https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html[data tiers] and machine learning nodes.
+
+[float]
+[id="{p}-enable"]
+== Enable autoscaling
+
+To enable autoscaling on an Elasticsearch cluster, you need to define one or more autoscaling policies. Each autoscaling policy applies to one or more NodeSets which share the same set of roles specified in the `node.roles` setting in the Elasticsearch configuration.
+
+[float]
+[id="{p}-{page_id}-policies"]
+=== Define autoscaling policies
+
+Autoscaling policies can be defined in the `elasticsearch.alpha.elastic.co/autoscaling-spec` annotation of the Elasticsearch resource. Each autoscaling policy must have the following fields:
+
+* `name` is a unique name used to identify the autoscaling policy.
+* `roles` contains a set of node roles, unique across all the autoscaling policies, used to identify the NodeSets to which this policy applies. At least one NodeSet with the exact same set of roles must exist in the Elasticsearch resource specification.
+* `resources` helps define the minimum and maximum compute resources usage:
+** `nodeCount` defines the minimum and maximum nodes allowed in the tier.
+** `cpu` and `memory` enforce minimum and maximum compute resources usage for the Elasticsearch container.
+** `storage` enforces minimum and maximum storage request per PersistentVolumeClaim.
+
+[source,json]
+----
+{
+    "policies": [{
+      "name": "di",
+      "roles": ["data", "ingest" , "transform"],
+      "resources": {
+          "nodeCount": { "min": 3, "max": 8 },
+          "cpu": { "min": 2, "max": 8 },
+          "memory": { "min": "2Gi", "max": "16Gi" },
+          "storage": { "min": "64Gi", "max": "512Gi" }
+      }
+    },
+    {
+      "name": "ml",
+      "roles": ["ml"],
+      "resources": {
+          "nodeCount": { "min": 1, "max": 9 },
+          "cpu": { "min": 1, "max": 4 },
+          "memory": { "min": "2Gi", "max": "8Gi" }
+      }
+    }]
+}
+----
+
+WARNING: A node role should not be referenced in more than one autoscaling policy.
```

Comment:
Warning implies that something "bad" happens if users do not heed the warning. But in this case iiuc we are 
1. validating that there is not overlap in policies
2. even if the webhook is disabled we recheck in the reconciliation loop and issue events

So maybe a `NOTE` would be more appropriate?

Reply:
There is no validation, it might be a temporary situation in the case of a transition between different tier models for example (e.g. from co-located roles to dedicated nodes per role).
For now, when detected, [this situation is reported in the logs and in the autoscaling status](https://github.com/elastic/cloud-on-k8s/pull/4265).